src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageToolWindowDisplayService.kt

```diff
@@ -1,47 +1,53 @@
 package nl.tudelft.ewi.se.ciselab.testgenie.services
 
-import com.intellij.ui.components.JBLabel
+import com.intellij.ui.table.JBTable
 import com.intellij.util.ui.FormBuilder
-import java.awt.Font
-import javax.swing.JLabel
 import javax.swing.JPanel
+import javax.swing.table.AbstractTableModel
 
 class CoverageToolWindowDisplayService {
-    var mainPanel: JPanel ?= null
-    var panelTitleAbsolute = JLabel("Absolute Test Coverage")
-    var panelTitleRelative = JLabel("Relative Test Coverage")
-    var absoluteLines = JBLabel("Amount of Lines covered: " + " Total: ")
-    var absoluteBranch = JBLabel("Amount of Branches covered: " + " Total: ")
-    var absoluteMutant = JBLabel("Amount of Mutants covered: "  + " Total: ")
-    var relativeLines = JBLabel("Percentage of Lines covered: ")
-    var relativeBranch = JBLabel("Percentage of Branches covered: ")
-    var relativeMutant = JBLabel("Percentage of Mutants covered: ")
+    var mainPanel: JPanel? = null
+    var data: ArrayList<String> = arrayListOf("Coverage", "Lines", "Branches", "Mutants", "", "", "", "")
```

Comment:
Why do you need those empty strings?

Reply:
They are for the second row of the table. I basically overwrite those entries when tests are generated.